core/Property/PropertySettings.php

```diff
@@ -0,0 +1,101 @@
+<?php
+/**
+ * Piwik - free/libre analytics platform
+ *
+ * @link http://piwik.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ *
+ */
+namespace Piwik\Property;
+
+use Piwik\Db;
+use Piwik\Piwik;
+use Piwik\Plugin\Settings;
+use Piwik\Property\Settings\Storage;
+use Piwik\Settings\Setting;
+use Piwik\Type\Type;
+
+class PropertySettings extends Settings
+{
+
+    /**
+     * @var int
+     */
+    private $idSite = null;
+
+    /**
+     * @var string
+     */
+    private $idType = null;
+
+    /**
+     * @param int $idSite The id of a site. If you want to get settings for a not yet created site just pass an empty value ("0")
+     * @param string $idType If no typeId is given, the type of the site will be used.
+     *
+     * @throws \Exception
+     */
+    public function __construct($idSite, $idType)
+    {
+        $this->idSite = $idSite;
+        $this->idType = $idType;
+        $this->storage = new Storage(Db::get(), $this->idSite);
+        $this->pluginName = 'PropertySettings';
+
+        $this->init();
+    }
+
+    protected function init()
+    {
+        $type = Type::getType($this->idType);
+        $type->configurePropertySettings($this);
+
+        /**
+         * This event is posted when generating settings for a property (website). You can add any property settings
+         * that you wish to be shown in the property manager (websites manager). If you need to add settings only for
+         * eg MobileApp properties you can use eg `$type->getId() === Piwik\Plugins\MobileAppType\Type::ID` and add only
+         * settings if the condition is true.
+         *
+         * @since Piwik 2.14.0
+         * @deprecated will be removed in Piwik 3.0.0
+         *
+         * @param PropertySettings $this
+         * @param \Piwik\Type\Type $type
+         * @param int $idSite
+         */
+        Piwik::postEvent('Property.initPropertySettings', array($this, $type, $this->idSite));
```

Comment:
Plugins will be able to specify settings too. For example we will need this in a plugin for Apple App Store username & password setting but also for #8021